AsyncDisplayKit/Details/ASPhotosFrameworkImageRequest.m

```diff
@@ -0,0 +1,169 @@
+//
+//  ASPhotosFrameworkImageRequest.m
+//  AsyncDisplayKit
+//
+//  Created by Adlai Holler on 9/25/15.
+//  Copyright © 2015 Facebook. All rights reserved.
+//
+
+#import "ASPhotosFrameworkImageRequest.h"
+#import "ASBaseDefines.h"
+
+NSString *const ASPhotosURLScheme = @"ph";
+
+static NSString *const _ASPhotosURLQueryKeyWidth = @"width";
+static NSString *const _ASPhotosURLQueryKeyHeight = @"height";
+
+// value is PHImageContentMode value
+static NSString *const _ASPhotosURLQueryKeyContentMode = @"contentmode";
+
+// value is PHImageRequestOptionsResizeMode value
+static NSString *const _ASPhotosURLQueryKeyResizeMode = @"resizemode";
+
+// value is PHImageRequestOptionsDeliveryMode value
+static NSString *const _ASPhotosURLQueryKeyDeliveryMode = @"deliverymode";
+
+// value is PHImageRequestOptionsVersion value
+static NSString *const _ASPhotosURLQueryKeyVersion = @"version";
+
+// value is 0 or 1
+static NSString *const _ASPhotosURLQueryKeyAllowNetworkAccess = @"network";
+
+static NSString *const _ASPhotosURLQueryKeyCropOriginX = @"crop_x";
+static NSString *const _ASPhotosURLQueryKeyCropOriginY = @"crop_y";
+static NSString *const _ASPhotosURLQueryKeyCropWidth = @"crop_w";
+static NSString *const _ASPhotosURLQueryKeyCropHeight = @"crop_h";
+
+@implementation ASPhotosFrameworkImageRequest
```

Comment:
OK here's a crazy idea: make this intermediate object private, and publicly expose just a category on NSURL with an initializer that mirrors the PHImageManager image request method:

 ```objectivec
// PHImageManager request method is:
- (PHImageRequestID)requestImageForAsset:(PHAsset * _Nonnull)asset
                              targetSize:(CGSize)targetSize
                             contentMode:(PHImageContentMode)contentMode
                                 options:(PHImageRequestOptions *)options
                           resultHandler:(void (^)(UIImage *result, NSDictionary *info))resultHandler;

// So we publicly expose:

@interface NSURL (ASPhotosFrameworkURLs)
// could also be +as_URLWithAsset:…
-(instancetype)as_initWithAssetLocalIdentifier:(NSString *)assetLocalIdentifier
									targetSize:(CGSize)targetSize
								   contentMode:(PHImageContentMode)contentMode
									   options:(PHImageRequestOptions *)options;
@end
```

This way the user is very clearly communicating what they want ASMultiplexImageNode to do in order to get the image, and it leaves our URLforImageIdentifier API clean.